airflow/providers/google/cloud/operators/cloud_batch.py

```diff
@@ -0,0 +1,327 @@
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Optional, Sequence, Union
+
+from google.api_core import operation  # type: ignore
+from google.cloud.batch_v1 import Job, Task
+
+from airflow.exceptions import AirflowException
+from airflow.models.baseoperator import BaseOperator
+from airflow.providers.google.cloud.hooks.cloud_batch import CloudBatchHook
+from airflow.providers.google.cloud.triggers.cloud_batch import \
+    CloudBatchJobFinishedTrigger
+from airflow.utils.context import Context
+
+
+class CloudBatchSubmitJobOperator(BaseOperator):
+    """
+    Submit a job and wait for its completion.
+
+    :param project_id: Required. The ID of the Google Cloud project that the service belongs to.
+    :param region: Required. The ID of the Google Cloud region that the service belongs to.
+    :param job_name: Required. The name of the job to create.
+    :param job: Required. The job descriptor containing the configuration of the job to submit.
+    :param polling_period_seconds: Optional: Control the rate of the poll for the result of deferrable run.
+        By default, the trigger will poll every 10 seconds.
+    :param timeout: The timeout for this request.
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+         Service Account Token Creator IAM role to the directly preceding identity, with first
+         account from the list granting this role to the originating account (templated).
+    :param deferrable: Run operator in the deferrable mode
+
+    """
+
+    template_fields = (
+        'project_id',
+        'region',
+        'gcp_conn_id',
+        'impersonation_chain',
+        'job_name'
+    )
+
+    def __init__(
+        self,
+        project_id: str,
+        region: str,
+        job_name: str,
+        job: Job,
+        polling_period_seconds: float = 10,
+        timeout_seconds: Union[float, None] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        deferrable: bool = False,
```

Comment:
Comment to self: deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False)

Run pre-commit hook